net/openvswitch/files/openvswitch.init

```diff
@@ -90,29 +95,41 @@ ovs_xx() {
 	esac
 }
 
-ovs_bridge_parse_port() {
-	case "$1" in
-		*:*)
-			port="${1%%:*}"
-			type="${1#*:}"
-		;;
-		*)
-			port="$1"
-			type=""
-		;;
-	esac
-}
-
 ovs_bridge_port_add() {
-	[ -n "$1" ] || return
+	local cfg="$1"
+
+	local bridge disabled ofport port tag type
+
+	config_get_bool disabled "$cfg" disabled 0
+	[ "$disabled" = "0" ] || return
+
+	config_get bridge "$cfg" bridge
+	ovs-vsctl br-exists ovsw0 || return
+
+	config_get port "$cfg" port
+	[ -n "$port" ] || return
+
+	config_get ofport "$cfg" ofport
+
+	config_get tag "$cfg" tag
+	if [ -n "$tag" ]; then
+		if cur_tag="$(ovs-vsctl get port "$port" tag 2>/dev/null)"; then
+			[ "$tag" = "$cur_tag" ] || __del_port=1
+		fi
+	fi
+
+	config_get type "$cfg" type
+	if [ -n "$type" ]; then
+		if cur_type="$(ovs-vsctl get interface "$port" type 2>/dev/null)"; then
```

Comment:
`cur_type` should be declared as `local`

Reply:
Done. Also for cur_tag, and del_port.